pkg/operator/ceph/csi/spec.go

```diff
@@ -291,16 +288,18 @@ func startDrivers(clientset kubernetes.Interface, namespace string, ver *version
 		tp.RBDPluginUpdateStrategy = rollingUpdate
 	}
 
-	if ver.Major > KubeMinMajor || (ver.Major == KubeMinMajor && ver.Minor < provDeploymentSuppVersion) {
+	logger.Infof("Kubernetes version is %s.%s", ver.Major, ver.Minor)
+	if ver.Major < KubeMinMajor || (ver.Major == KubeMinMajor && ver.Minor < provDeploymentSuppVersion) {
```

Comment:
I don't see this making a difference in the behavior. `ver.Major` is always `1` when we're deploying rook. Rook basically only supports `1.11 - 1.18`. 
```suggestion
	if ver.Major == KubeMinMajor && ver.Minor < provDeploymentSuppVersion {
```

Reply:
On what version of K8s did we see the issue where the stateful set was deployed? Or do we have any other clues?

Reply:
In ocp 4.6 the kubernetes version was 4.6 due to that we have seen this issue

Reply:
I see now, it was the OCP version. How about if we go with my suggested change? No need to check for `< KubeMinMajor`, we can just check for equality.

Reply:
sure doing it now